### DISCUSSION TITLE
Come posso ottenere le fatture emesse in un certo range di date?

### DISCUSSION THREAD

**Discussion by sensorario** [fattureincloud/api#136] [Q&A] [answered]
Tramite l'API `GET /c/{company_id}/issued_documents` ottengo tutte le fatture. Non posso filtrare per data. Avrei nevessità di poter interrogare in questo modo. L'alternativa sarebbe quella di scorrere tutte le fatture, eventualmente salvate in una cache, e raggrupparle per data. Mentre risolvo il problema in questo modo, .. spero qualcuno legga questa discussione e mi suggerisca eventualmente soluzioni alternative.

**Comment by valmoz** [comment, marked as answer]
Ciao @sensorario 
dalla [documentazione dei filtri](https://developers.fattureincloud.it/docs/basics/additional-capabilities/filter-results) io vedo che sulla ListIssuedDocument il campo "date" è filterable... Ti riferisci forse a un'altro campo?

Grazie
Mauro

**Reply by sensorario** [reply]
Io sto guardando qui. Non vedo il campo date. https://developers.fattureincloud.it/api-reference/#get-/c/-company_id-/issued_documents

**Reply by sensorario** [reply]
Ah! Sto usando la v2 delle api.

**Reply by valmoz** [reply]
Certo, ho compreso ;)

I filtri utilizzano un campo unico, chiamato "q", che accetta in ingresso delle clausole SQL-like che funzionano un po' come il campo WHERE di una normale query.
Nel tuo caso, "q" potrebbe avere un valore simile a:
`date = SEARCHED_DATE`

Se non stai utilizzando le nostre SDK per l'implementazione, allora il campo "q" deve essere inserito in querystring. Altrimenti, le SDK sono equipaggiate con un [Filter Helper](https://developers.fattureincloud.it/docs/sdks/additional-capabilities/filter-results) che permette di semplificare la scrittura delle condizioni per il filtraggio.

Trovi tutta la documentazione relativa al parametro "q" nel link che ti ho girato in precedenza.

**Reply by sensorario** [reply]
Ottimo, ... grazie della risposta!
